pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -193,7 +194,31 @@ func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.Statef
 			errs = append(errs, fmt.Errorf("Failed to retrieve PVC %s: %s", claim.Name, err))
 			spc.recordClaimEvent("create", set, pod, &claim, err)
 		}
-		// TODO: Check resource requirements and accessmodes, update if necessary
+		// Check resource requirements and accessmodes, update if necessary
+		if err == nil {
+			change := false
+			if len(claim.Spec.AccessModes) > 0 {
```

Comment:
What if `len(claim.Spec.AccessModes)=0` and `len(currentClaim.Spec.AccessModes)>0` ?

Need to adjust the conditions.

Reply:
We only check the condition of `len(claim.Spec.AccessModes)>0`, if the length is zero, we do nothing.